crates/cxx-qt-lib/src/gui/qguiapplication.rs

```diff
@@ -0,0 +1,161 @@
+        fn qapplicationOrganizationDomain(app: &QGuiApplication) -> QString;
+        #[doc(hidden)]
+        #[rust_name = "qguiapplication_organization_name"]
+        fn qapplicationOrganizationName(app: &QGuiApplication) -> QString;
+        #[doc(hidden)]
+        #[rust_name = "qguiapplication_set_application_name"]
+        fn qapplicationSetApplicationName(app: Pin<&mut QGuiApplication>, name: &QString);
+        #[doc(hidden)]
+        #[rust_name = "qguiapplication_set_application_version"]
+        fn qapplicationSetApplicationVersion(app: Pin<&mut QGuiApplication>, version: &QString);
+        #[doc(hidden)]
+        #[rust_name = "qguiapplication_set_library_paths"]
+        fn qapplicationSetLibraryPaths(app: Pin<&mut QGuiApplication>, paths: &QStringList);
+        #[doc(hidden)]
+        #[rust_name = "qguiapplication_set_organization_domain"]
+        fn qapplicationSetOrganizationDomain(app: Pin<&mut QGuiApplication>, domain: &QString);
+        #[doc(hidden)]
+        #[rust_name = "qguiapplication_set_organization_name"]
+        fn qapplicationSetOrganizationName(app: Pin<&mut QGuiApplication>, name: &QString);
+    }
+
+    impl UniquePtr<QGuiApplication> {}
+}
+
+pub use ffi::QGuiApplication;
+
+impl QGuiApplication {
+    /// Prepends path to the beginning of the library path list,
+    /// ensuring that it is searched for libraries first.
+    /// If path is empty or already in the path list, the path list is not changed.
+    pub fn add_library_path(self: Pin<&mut Self>, path: &QString) {
+        ffi::qguiapplication_add_library_path(self, path);
+    }
+
+    /// The name of this application
+    pub fn application_name(&self) -> QString {
+        ffi::qguiapplication_application_name(self)
+    }
+
+    /// The version of this application
+    pub fn application_version(&self) -> QString {
+        ffi::qguiapplication_application_version(self)
+    }
+
+    /// Enters the main event loop and waits until exit() is called,
+    /// and then returns the value that was set to exit() (which is 0 if exit() is called via quit()).
+    pub fn exec(self: Pin<&mut Self>) -> i32 {
+        ffi::qguiapplication_exec(self)
+    }
+
+    /// Returns a list of paths that the application will search when dynamically loading libraries.
+    pub fn library_paths(&self) -> QStringList {
+        ffi::qguiapplication_library_paths(self)
+    }
+
+    /// Initializes the window system and constructs an application object with command line arguments in args.
+    pub fn new(args: Vec<String>) -> cxx::UniquePtr<Self> {
+        ffi::qguiapplication_new(args)
+    }
+
+    /// Initializes the window system and constructs an application object with command line arguments in args.
+    pub fn new_from_args_os(args: std::env::ArgsOs) -> cxx::UniquePtr<Self> {
+        Self::new(
+            args.map(|string| {
+                // OsStrings needs to be converted into UTF-8 Strings
+                string.to_string_lossy().to_string()
```

Comment:
Yup, I'll see if i can get it back to c_char maybe by reusing our QVector, as it needs to be a type that CXX supports...

Reply:
Changed this so we are back to similar code as before except it construct them into a `QVector<QByteArray>` which can then be passed across the CXX bridge and it can be owned in the QObject which means it can live long enough for the lifetime of the QGuiApplication.